pkg/controllers/member/serviceexport/controller.go

```diff
@@ -256,6 +268,70 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *corev1.Service, export *fleetnetv1alpha1.InternalServiceExport) error {
+	export.Spec.Type = service.Spec.Type
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+	export.Spec.IsInternalLoadBalancer = service.Annotations[objectmeta.AzureLoadBalancerInternalAnnotation] == "true"
+	if export.Spec.IsInternalLoadBalancer {
+		// no need to populate the PublicIPResourceID and IsDNSLabelConfigured which are only applicable for external load balancer
+		return nil
+	}
+	id, err := r.fetchPublicIPResourceID(ctx, service)
+	if err != nil {
+		return err
+	}
+	if id == nil || len(*id) == 0 {
+		return fmt.Errorf("public ip resource id %v has not been configured in the load balancer yet", id)
+	}
+	export.Spec.PublicIPResourceID = id
+	// Note the user can set the dns label via the Azure portal or Azure CLI without updating service.
+	// This information may be stale as we don't monitor the public IP address resource.
+	export.Spec.IsDNSLabelConfigured, err = r.isDNSLabelConfigured(ctx, *id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) fetchPublicIPResourceID(ctx context.Context, service *corev1.Service) (*string, error) {
+	serviceKObj := klog.KObj(service)
+	lb, err := r.AzureLoadBalancerClient.Get(ctx, r.LoadBalancerResourceGroupName, publicLoadBalancerName, nil)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get Azure load balancer", "service", serviceKObj)
+		return nil, err
+	}
+	frontedIPName := "a" + string(service.UID)
+	if lb.Properties != nil {
+		for _, frontendIPConfig := range lb.Properties.FrontendIPConfigurations {
+			if frontendIPConfig.Name != nil && *frontendIPConfig.Name == frontedIPName {
```

Comment:
hmmmm, sorry but I realized one thing: services many share frontends. Cx can create multiple services just specifying the same public IP. In this case, these services will share the same frontend named after the first service obj UID. See https://github.com/kubernetes-sigs/cloud-provider-azure/blob/6c002772ad81dc56e74a3a726a3b831387989560/pkg/provider/azure_loadbalancer.go#L4113 

To resolve this case, we need to list and find the PIP directly. Do you think if we keep this for now given that sharing frontend is a less common usage or we directly turn to list PIP and find the PIP resource according to the public IP addr in the svc status? In this case, we can get rid of the loadbalancer at all.